hooks/useSubmitVote.ts

```diff
@@ -100,9 +101,75 @@ export const useSubmitVote = () => {
     }
   )
 
+  const { error: multiChoiceError, loading: multiChoiceLoading, execute: multiChoiceExecute } = useAsyncCallback(
+    async ({
+      vote,
+      voterTokenRecord,
+      voteWeights,
+      comment,
+    }: {
+      vote: VoteKind
+      voterTokenRecord: ProgramAccount<TokenOwnerRecord>
+      voteWeights: number[]
+      comment?: string
+    }) => {
+      const rpcContext = new RpcContext(
+        proposal!.owner,
+        getProgramVersionForRealm(realmInfo!),
+        wallet!,
+        connection.current,
+        connection.endpoint
+      )
+
+      const msg = comment
+        ? new ChatMessageBody({
+            type: ChatMessageBodyType.Text,
+            value: comment,
+          })
+        : undefined
+
+      const confirmationCallback = async () => {
+        await queryClient.invalidateQueries(
+          voteRecordQueryKeys.all(connection.cluster)
+        )
+      }
+
+      try {
+        await castVote(
+          rpcContext,
+          realm!,
+          proposal!,
+          voterTokenRecord,
+          vote,
+          true,
+          msg,
+          client,
+          confirmationCallback,
+          voteWeights
+        )
+        queryClient.invalidateQueries({
```

Comment:
my understanding is that this won't actually do anything, you have to invalidate `proposalQueryKeys.all(connection.rpcEndpoint)` or whatever. 

HOWEVER, it shouldn't be needed either, because the proposal is a writeable account in the transaction and therefore is auto-invalidated. so you can just delete this.

Reply:
oh I see now that you copypasted. yeah in any case you can delete this line.

Reply:
probably you should just delete this and make `voteWeights` an optional argument for the existing callback, and remove the `isMulti` parameter from `castVote`

Reply:
Done. Deleted the new callbacks and added optional `voteWeights` in the original callback.